ddl/placement_policy_test.go

```diff
@@ -974,6 +990,91 @@ func (s *testDBSuite6) TestDatabasePlacement(c *C) {
 	))
 }
 
+func (s *testDBSuite6) TestDropDatabaseGCPlacement(c *C) {
+	clearAllBundles(c)
+	failpoint.Enable("github.com/pingcap/tidb/store/gcworker/ignoreDeleteRangeFailed", `return`)
+	defer func() {
+		failpoint.Disable("github.com/pingcap/tidb/store/gcworker/ignoreDeleteRangeFailed")
+	}()
+
+	tk := testkit.NewTestKit(c, s.store)
+	tk.MustExec("drop database if exists db2")
+
+	tk.MustExec("use test")
+
+	tk.MustExec("create table t (id int) primary_region='r0' regions='r0'")
+	defer tk.MustExec("drop table if exists t")
+
+	tk.MustExec("create database db2")
+	tk.MustExec("create table db2.t0 (id int)")
+	tk.MustExec("create table db2.t1 (id int) primary_region='r1' regions='r1,r2'")
+	tk.MustExec(`create table db2.t2 (id int) primary_region='r1' regions='r1,r2' PARTITION BY RANGE (id) (
+        PARTITION p0 VALUES LESS THAN (100) primary_region='r2' regions='r2',
+        PARTITION p1 VALUES LESS THAN (1000)
+	)`)
+
+	is := tk.Se.GetInfoSchema().(infoschema.InfoSchema)
+	t, err := is.TableByName(model.NewCIStr("test"), model.NewCIStr("t"))
+	c.Assert(err, IsNil)
+
+	tk.MustExec("drop database db2")
+
+	bundles, err := infosync.GetAllRuleBundles(context.TODO())
+	c.Assert(err, IsNil)
+	c.Assert(len(bundles), Equals, 4)
+
+	gcWorker, err := gcworker.NewMockGCWorker(s.store)
+	c.Assert(err, IsNil)
+	c.Assert(gcWorker.DeleteRanges(context.TODO(), math.MaxInt64), IsNil)
```

Comment:
I remembered that gc will not handle `dbInfo.ID`, does that mean `policy of db` will not be deleted?

Reply:
I'm completing the missing cases  for https://github.com/pingcap/tidb/pull/28687 . But I think the reasons is:

1. Deleting rules when dropping may cause the reschedule of PD. I'm not sure if it'll introduce some extra cost.
2. Users may still expect their data fits the placement rule, even it is dropped. For example: compliance
3. Users can still access deleted data even if it is deleted like stale read. At this time they still want the data fits the placement rule for some reason like isolation.